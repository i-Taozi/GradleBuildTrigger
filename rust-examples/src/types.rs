///类型
pub fn types() {
    let decimal = 65.4321_f32;
    //不支持隐式转化
    //let integer: u8 = decimal;
    // FIXME ^ Comment out this line

    //显示转化
    let integer = decimal as u8;
    let character = integer as char;

    println!("Casting: {} -> {} -> {}", decimal, integer, character);

    // 1000 already fits in a u16
    println!("1000 as a u16 is: {}", 1000 as u16);

    // 1000 - 256 - 256 - 256 = 232
    // Under the hood, the first 8 least significant bits (LSB) are kept,
    // while the rest towards the most significant bit (MSB) get truncated.
    //println!("1000 as a u8 is : {}", 1000 as u8);
    // -1 + 256 = 255
    println!("  -1 as a u8 is : {}", (-1i8) as u8);
    println!("1000 mod 256 is : {}", 1000 % 256);

    println!(" 128 as a i16 is: {}", 128 as i16);
    //println!(" 128 as a i8 is : {}", 128 as i8);
    //println!("1000 as a u8 is : {}", 1000 as u8);
    //println!(" 232 as a i8 is : {}", 232 as i8);

    literals();
}

fn literals() {
    //类型作为后缀
    let x = 1u8;
    let y = 2u32;
    let z = 3f32;

    //非固定文字，其类型取决于如何使用
    let i = 1;
    let f = 1.0;

    //Returns the size of the pointed-to value in bytes.
    println!("size of `x` in bytes: {}", std::mem::size_of_val(&x));
    println!("size of `y` in bytes: {}", std::mem::size_of_val(&y));
    println!("size of `z` in bytes: {}", std::mem::size_of_val(&z));
    println!("size of `i` in bytes: {}", std::mem::size_of_val(&i));
    println!("size of `f` in bytes: {}", std::mem::size_of_val(&f));
}
