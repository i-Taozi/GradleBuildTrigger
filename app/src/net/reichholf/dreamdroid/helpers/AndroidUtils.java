package net.reichholf.dreamdroid.helpers;

import android.content.Context;
import android.content.pm.PackageManager;
import android.view.View;
import android.widget.TextView;

public class AndroidUtils {
	public static boolean isLeanback(Context ctx) {
		return (ctx.getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEVISION)
				|| ctx.getPackageManager().hasSystemFeature(PackageManager.FEATURE_LEANBACK));
	}

	public static void setTextOrHide(TextView textView, String text) {
		if (text.isEmpty())
			textView.setVisibility(View.GONE);
		else
			textView.setText(text);
	}
}
