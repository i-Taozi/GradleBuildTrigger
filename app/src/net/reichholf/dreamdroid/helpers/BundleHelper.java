/* Â© 2010 Stephan Reichholf <stephan at reichholf dot net>
 * 
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers;

import java.util.ArrayList;

/**
 * @author sre
 *
 */
public class BundleHelper {
	public static ArrayList<String> toStringArrayList(CharSequence[] strings){
		ArrayList<String> list = new ArrayList<>();
		for (CharSequence string : strings) {
			list.add(string.toString());
		}
		return list;
	}
	
	public static CharSequence[] toCharSequenceArray(ArrayList<String> strings){
		CharSequence[] list = new CharSequence[strings.size()];
		for(int i = 0; i < strings.size(); ++i){
			list[i] = strings.get(i);
		}
		return list;
	}
}
