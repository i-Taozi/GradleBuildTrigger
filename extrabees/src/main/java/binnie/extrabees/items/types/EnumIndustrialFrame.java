package binnie.extrabees.items.types;

import forestry.api.apiculture.IBeeGenome;
import forestry.api.apiculture.IBeeModifier;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;

import javax.annotation.Nullable;

public enum EnumIndustrialFrame implements IBeeModifier {

	EMPTY("Empty", 5, 0),
	LIGHT("Glowstone Lighting", 2, 4) {
		@Override
		protected void init() {
			lighted = true;
		}
	},
	RAIN("Rain Shielding", 2, 4) {
		@Override
		protected void init() {
			rain = true;
		}
	},
	SUNLIGHT("Sunlight Simulator", 4, 8) {
		@Override
		protected void init() {
			lighted = true;
			sunlight = true;
		}
	},
	SOUL("Low Grade Mutagen", 5, 15) {
		@Override
		protected void init() {
			mutationMod = 1.3f;
		}
	},
	URANIUM("High Grade Mutagen", 10, 50) {
		@Override
		protected void init() {
			mutationMod = 2.0f;
		}
	},
	CAGE("Meshed Restrainer", 3, 12) {
		@Override
		protected void init() {
			territoryMod = 0.4f;
		}
	},
	FREEDOM("Territory Extension", 3, 16) {
		@Override
		protected void init() {
			territoryMod = 1.4f;
		}
	},
	HONEY("Honey Amplifier", 4, 12) {
		@Override
		protected void init() {
			productionMod = 1.4f;
		}
	},
	JELLY("Gelatin Amplifier", 8, 36) {
		@Override
		protected void init() {
			productionMod = 1.8f;
		}
	},
	LEAF("Pollinator MK I", 3, 15) {
		@Override
		protected void init() {
			floweringMod = 1.4f;
		}
	},
	POLLEN("Pollinator MK II", 7, 25) {
		@Override
		protected void init() {
			floweringMod = 2.0f;
		}
	},
	CLAY("Lifespan Extensor", 2, 10) {
		@Override
		protected void init() {
			lifespanMod = 1.4f;
		}
	},
	EMERALD("Eon Simulator", 7, 20) {
		@Override
		protected void init() {
			lifespanMod = 2.0f;
		}
	},
	NETHER_STAR("Immortality Gate", 12, 50) {
		@Override
		protected void init() {
			lifespanMod = 20.0f;
		}
	},
	POISON("Mortality Inhibitor", 8, 18) {
		@Override
		protected void init() {
			lifespanMod = 0.5f;
		}
	};

	private final String name;
	protected float territoryMod;
	protected float mutationMod;
	protected float lifespanMod;
	protected float productionMod;
	protected float floweringMod;
	protected boolean lighted;
	protected boolean sunlight;
	protected boolean rain;
	private final int wearMod;
	private final int power;

	EnumIndustrialFrame(final String name, final int wear, final int power) {
		this.territoryMod = 1.0f;
		this.mutationMod = 1.0f;
		this.lifespanMod = 1.0f;
		this.productionMod = 1.0f;
		this.floweringMod = 1.0f;
		this.lighted = false;
		this.sunlight = false;
		this.rain = false;
		this.name = name;
		this.wearMod = wear;
		this.power = power;
		init();
	}

	protected void init() {

	}

	public static ItemStack getItemStack(final Item item, final EnumIndustrialFrame frame) {
		final ItemStack stack = new ItemStack(item);
		final NBTTagCompound nbt = new NBTTagCompound();
		nbt.setInteger("frame", frame.ordinal());
		stack.setTagCompound(nbt);
		return stack;
	}

	@Override
	public float getTerritoryModifier(final IBeeGenome genome, final float currentModifier) {
		return this.territoryMod;
	}

	@Override
	public float getMutationModifier(final IBeeGenome genome, final IBeeGenome mate, final float currentModifier) {
		return this.mutationMod;
	}

	@Override
	public float getLifespanModifier(final IBeeGenome genome, @Nullable final IBeeGenome mate, final float currentModifier) {
		return this.lifespanMod;
	}

	@Override
	public float getProductionModifier(final IBeeGenome genome, final float currentModifier) {
		return this.productionMod;
	}

	@Override
	public float getFloweringModifier(final IBeeGenome genome, final float currentModifier) {
		return this.floweringMod;
	}

	@Override
	public boolean isSealed() {
		return this.rain;
	}

	@Override
	public boolean isSelfLighted() {
		return this.lighted;
	}

	@Override
	public boolean isSunlightSimulated() {
		return this.sunlight;
	}

	@Override
	public boolean isHellish() {
		return false;
	}

	public String getName() {
		return this.name;
	}

	public int getWearModifier() {
		return this.wearMod;
	}

	public int getPowerUsage() {
		return this.power;
	}

	@Override
	public float getGeneticDecay(final IBeeGenome genome, final float currentModifier) {
		return 1.0f;
	}

}
