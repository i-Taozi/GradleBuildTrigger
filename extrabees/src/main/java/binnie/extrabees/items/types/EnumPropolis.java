package binnie.extrabees.items.types;

import binnie.core.util.I18N;
import binnie.extrabees.modules.ModuleCore;
import binnie.extrabees.utils.Utils;
import forestry.api.recipes.RecipeManagers;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fluids.FluidStack;

public enum EnumPropolis implements IEBEnumItem {
	WATER(2405321, 12762791, "Water"),
	OIL(1519411, 12762791, "oil"),
	FUEL(10718482, 12762791, "fuel"),
	MILK,
	FRUIT,
	SEED,
	ALCOHOL,
	CREOSOTE(8877313, 12428819, "creosote"),
	GLACIAL,
	PEAT;

	private final int primaryColor;
	private final int secondaryColor;
	private final String liquidName;
	private boolean active;

	EnumPropolis() {
		this(16777215, 16777215, "");
		this.active = false;
	}

	EnumPropolis(int primaryColor, int secondaryColor, String liquidName) {
		this.active = true;
		this.primaryColor = primaryColor;
		this.secondaryColor = secondaryColor;
		this.liquidName = liquidName;
	}

	public static EnumPropolis get(final ItemStack itemStack) {
		final int i = itemStack.getItemDamage();
		if (i >= 0 && i < values().length) {
			return values()[i];
		}
		return values()[0];
	}

	public void addRecipe() {
		final FluidStack liquid = Utils.getFluidFromName(this.liquidName, 500);
		if (liquid != null) {
			RecipeManagers.squeezerManager.addRecipe(20, this.get(1), liquid, ItemStack.EMPTY, 0);
		}
	}

	public int getSpriteColour(int renderPass) {
		if (renderPass == 0) {
			return primaryColor;
		}
		if (renderPass == 1) {
			return secondaryColor;
		}
		return 0xffffff;
	}

	@Override
	public boolean isActive() {
		return this.active && Utils.getFluidFromName(this.liquidName, 100) != null;
	}

	@Override
	public ItemStack get(final int amount) {
		return new ItemStack(ModuleCore.propolis, amount, this.ordinal());
	}

	@Override
	public String getName(final ItemStack itemStack) {
		return I18N.localise("extrabees.item.propolis." + this.name().toLowerCase());
	}

}
