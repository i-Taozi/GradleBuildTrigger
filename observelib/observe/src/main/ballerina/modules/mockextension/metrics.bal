// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
//
// WSO2 Inc. licenses this file to you under the Apache License,
// Version 2.0 (the "License"); you may not use this file except
// in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import ballerina/jballerina.java;

# Holds the key and value of tag.
#
public type Tag record {
  string key;
  string value;
};

# Holds parameters required to id a metric.
#
public type MetricId record {
  string name;
  string description;
  Tag[] tags;
};

# Holds percentile and value.
#
public type PercentileValue record {
  float percentile;
  float value;
};

# Holds a time duration.
#
public type TimeWindow record {
    int seconds;
    int nanos;
};

# Holds a snapshot of statistics.
#
public type Snapshot record {
  TimeWindow timeWindow;
  float min;
  float max;
  float mean;
  float stdDev;
  PercentileValue[] percentileValues;
};

# Holds counter related metrics data.
#
public type Counter record {
  MetricId id;
  int value;
};

# Holds gauge related metrics data.
#
public type Gauge record {
  MetricId id;
  float value;
  int count;
  float sum;
  Snapshot[] snapshots;
};

# Holds polled gauge related metrics data.
#
public type PolledGauge record {
  MetricId id;
  float value;
};

# Holds metrics data.
#
public type Metrics record {
  Counter[] counters;
  Gauge[] gauges;
  PolledGauge[] polledGauges;
};

# Get all the current metrics.
#
# + return - Current metrics
public isolated function getMetrics() returns Metrics {
    json metricsJson = externGetMetrics();
    Metrics | error metrics = metricsJson.cloneWithType(Metrics);
    if (metrics is error) {
        panic error("cannot convert to Metrics record; json string: " + metricsJson.toJsonString(), metrics);
    } else {
        return metrics;
    }
}

# Get all the current metrics.
#
# + return - The metrics currently in the metrics registry as a json
isolated function externGetMetrics() returns json = @java:Method {
    name: "getMetrics",
    'class: "org.ballerinalang.observe.mockextension.MetricsUtils"
} external;
