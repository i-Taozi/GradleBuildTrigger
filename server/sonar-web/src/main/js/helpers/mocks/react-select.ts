/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { GroupTypeBase, OptionProps } from 'react-select';

export function mockReactSelectOptionProps<
  OptionType,
  IsMulti extends boolean,
  GroupType extends GroupTypeBase<OptionType> = GroupTypeBase<OptionType>
>(
  data: OptionType,
  overrides?: OptionProps<OptionType, IsMulti, GroupType>
): OptionProps<OptionType, IsMulti, GroupType> {
  return {
    ...overrides,
    data
  } as OptionProps<OptionType, IsMulti, GroupType>;
}
