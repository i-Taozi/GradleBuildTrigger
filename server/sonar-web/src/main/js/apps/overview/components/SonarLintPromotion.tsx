/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { withCurrentUser } from '../../../components/hoc/withCurrentUser';
import SonarLintIcon from '../../../components/icons/SonarLintIcon';
import { translate } from '../../../helpers/l10n';
import { MetricKey } from '../../../types/metrics';
import { QualityGateStatusCondition } from '../../../types/quality-gates';

export interface SonarLintPromotionProps {
  currentUser: T.CurrentUser;
  qgConditions?: QualityGateStatusCondition[];
}

const CONDITIONS_TO_SHOW = [
  MetricKey.new_blocker_violations,
  MetricKey.new_critical_violations,
  MetricKey.new_info_violations,
  MetricKey.new_violations,
  MetricKey.new_major_violations,
  MetricKey.new_minor_violations,
  MetricKey.new_code_smells,
  MetricKey.new_bugs,
  MetricKey.new_vulnerabilities,
  MetricKey.new_security_rating,
  MetricKey.new_maintainability_rating,
  MetricKey.new_reliability_rating
];

export function SonarLintPromotion({ currentUser, qgConditions }: SonarLintPromotionProps) {
  const showMessage = qgConditions?.some(
    qgCondition =>
      CONDITIONS_TO_SHOW.includes(qgCondition.metric as MetricKey) && qgCondition.level === 'ERROR'
  );
  if (!showMessage || currentUser.usingSonarLintConnectedMode) {
    return null;
  }
  return (
    <div className="it__overview__sonarlint-promotion big-spacer-top overview-quality-gate-sonar-lint-info">
      <FormattedMessage
        id="overview.fix_failed_conditions_with_sonarlint"
        defaultMessage={translate('overview.fix_failed_conditions_with_sonarlint')}
        values={{
          link: (
            <>
              <a
                href="https://www.sonarqube.org/sonarlint/?referrer=sonarqube"
                rel="noopener noreferrer"
                target="_blank">
                SonarLint
              </a>
              <SonarLintIcon size={16} />
            </>
          )
        }}
      />
    </div>
  );
}

export default withCurrentUser(SonarLintPromotion);
