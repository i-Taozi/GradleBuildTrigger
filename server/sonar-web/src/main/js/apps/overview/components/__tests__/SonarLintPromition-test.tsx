/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { mockQualityGateStatusCondition } from '../../../../helpers/mocks/quality-gates';
import { mockCurrentUser } from '../../../../helpers/testMocks';
import { MetricKey } from '../../../../types/metrics';
import { SonarLintPromotion, SonarLintPromotionProps } from '../SonarLintPromotion';

it('should render correctly', () => {
  expect(shallowRender().type()).toBeNull();
  expect(
    shallowRender({ currentUser: mockCurrentUser({ usingSonarLintConnectedMode: true }) }).type()
  ).toBeNull();
  expect(
    shallowRender({
      qgConditions: [mockQualityGateStatusCondition({ metric: MetricKey.new_bugs, level: 'ERROR' })]
    })
  ).toMatchSnapshot('has failed condition');
});

it.each(
  [
    MetricKey.new_blocker_violations,
    MetricKey.new_critical_violations,
    MetricKey.new_info_violations,
    MetricKey.new_violations,
    MetricKey.new_major_violations,
    MetricKey.new_minor_violations,
    MetricKey.new_code_smells,
    MetricKey.new_bugs,
    MetricKey.new_vulnerabilities,
    MetricKey.new_security_rating,
    MetricKey.new_maintainability_rating,
    MetricKey.new_reliability_rating
  ].map(Array.of)
)('should show message for %s', metric => {
  const wrapper = shallowRender({
    qgConditions: [mockQualityGateStatusCondition({ metric: metric as string })]
  });
  expect(wrapper.type()).not.toBeNull();
});

function shallowRender(props: Partial<SonarLintPromotionProps> = {}) {
  return shallow(<SonarLintPromotion currentUser={mockCurrentUser()} {...props} />);
}
