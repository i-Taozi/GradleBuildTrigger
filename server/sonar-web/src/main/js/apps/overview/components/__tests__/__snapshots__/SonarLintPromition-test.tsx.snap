// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly: has failed condition 1`] = `
<div
  className="it__overview__sonarlint-promotion big-spacer-top overview-quality-gate-sonar-lint-info"
>
  <FormattedMessage
    defaultMessage="overview.fix_failed_conditions_with_sonarlint"
    id="overview.fix_failed_conditions_with_sonarlint"
    values={
      Object {
        "link": <React.Fragment>
          <a
            href="https://www.sonarqube.org/sonarlint/?referrer=sonarqube"
            rel="noopener noreferrer"
            target="_blank"
          >
            SonarLint
          </a>
          <SonarLintIcon
            size={16}
          />
        </React.Fragment>,
      }
    }
  />
</div>
`;
